Nos/Views/ReportMenu.swift

```diff
@@ -116,30 +122,90 @@ struct ReportMenuModifier: ViewModifier {
             }
         }
     }
+    
+    typealias TopLevelDisplayName = String
+    /// An enum to simplify the user selection through the sequence of connected
+    /// dialogs
+    enum UserSelection: Equatable {
+        case selected(ReportCategory, TopLevelDisplayName)
+        case sendToNos(ReportCategory)
+        case flagPublicly(ReportCategory)
+        
+        var displayName: String {
+            switch self {
+            case .selected(_, let displayName):
+                return displayName
+            case .sendToNos(let category), .flagPublicly(let category):
+                return category.displayName
+            }
+        }
 
-    /// Generates a series of Buttons for the given report categories.
-    func subCategoryButtons(for categories: [ReportCategory]) -> [ButtonState<ReportCategory>] {
-        categories.map { subCategory in
-            ButtonState(action: .send(subCategory)) {
-                TextState(verbatim: subCategory.displayName)
-            } 
+        var confirmationAlertMessage: String {
+            switch self {
+            case .sendToNos(let category):
+                return String(localized: .localizable.reportSendToNosConfirmation(category.displayName))
+            case .flagPublicly(let category):
+                return String(localized: .localizable.reportFlagPubliclyConfirmation(category.displayName))
+            case .selected(let category, _):
+                Log.error("Invalid .selected variant")
+                return String(localized: .localizable.reportFlagPubliclyConfirmation(category.displayName))
+            }
         }
     }
     
-    /// Publishes a report for the currently selected 
-    func publishReport() {
-        guard let keyPair = currentUser.keyPair,
-            let selectedCategory else {
+    /// List of the top-level report categories we care about
+    /// Out vocabulary is much bigger
+    func topLevelButtons() -> [ButtonState<UserSelection>] {
+        let spam = ("SP", "Spam")
```

Comment:
@mplorentz I refactored this to be able to integrate the new display names to the categories data.
I also took the freedom to update some mappings because I think they were wrong because the illegal category wasn't used but we were using the illegal text. Also the only category we have that maps to nip-56 profanity was "coarse language" but we wanted to present that as "harrasment or intolerance" (harassment is already part of illegal subcategories). So I change the label to "coarse language or intolerance" which I think maps better to something that although it not illegal, could be reported. This also avoids breaking the mapping when we show reports from other clients that were used with the intent of coarse language but nothing to do with intolerance or harassment.

So I have this now, 
<img width="388" alt="Screenshot 2024-04-24 at 2 12 37 PM" src="https://github.com/planetary-social/nos/assets/18184/9e0051c5-1b49-4530-8d24-3b4716a0c388">



which in order, map to the following categories:
1. spam (unchanged, mapped to nip 56 spam),
2. nudity (changed the localized string which used to be 'Nudity & Sex', mapped to nip 56 nudity),
3. coarseLanguage (changed the label from 'Harassment or Intolerance', mapped to nip 56 profanity),
4. illegal (used to be violence which is mapped to nip 56 other, now mapped to nip 56 illegal),
5. other (unchanged, mapped to nip 56 other)

Reply:
@setch-l ^ I took the freedom to do some changes in the category mapping to avoid breaking the categories we have in cleanstr and we can reuse the names without clashes when displaying them from other sources

Reply:
These look good to me!